### PR TITLE
Fix error validation for required `name` fields in submission

### LINKED ISSUE #1353: Empty required name input not displaying error message after saving
**Describe the bug**
Input fields of type "name" don't display an error message when left empty after saving the form in submission.

**To Reproduce**
- Setup a DSpace 7 with a submission form containing required input(s) of type "name" (doesn't matter if it's wrapped in a relation-field or not, both contain the bug)
- Go to submission and fill in a couple of fields, but leave the required name field empty
- Click Save
- The name field doesn't show any error messages. Other required fields do.

**Expected behavior**
The required name field should display an error message after clicking save and leaving it empty.

### PR DESCRIPTION
**Pull request by ybnd**
## References
* Fixes #1353

## Description
Currently, `name` fields in submission forms are not flagged as invalid when they return as a server error (i.e.: empty required field)
Came across three separate tiny issues while debugging & fixing this
* `FormBuilderService.findById` was expecting the `id` of one of the group's inputs to end in a numeric suffix. I haven't been able to track down where this extra suffix would be set, so I'm assuming this was a bug.
  IMO, for the field to be flagged "correctly", we should flag the _group_ itself so that _both_ inputs show an error
* When a whole group is marked as invalid, its inputs should also be marked invalid. As far as I'm aware this will only affect the case of invalidation via server errors.
* With these "concat fields", the individual inputs will _also_ get an error hint when invalid. Added a basic mechanism to disable them.

## Instructions for Reviewers
1. Evaluate whether the changes could have adverse effects on other submission forms/groups/fields/...

1. Confirm that the original issue is addressed; after the fixes in this PR
    * `name` fields should be flagged as invalid if required & left empty on save/deposit
    * when the field is invalid, the two inputs it contains should also be marked as invalid
    * the error hint should only be shown once, at the `FormGroup` level
    

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Review by tdonohue — APPROVED** [review]
👍 Thanks @ybnd !  I tested this today and it fixes the issue with validating the `name` field.  I attempted making several other field types required to verify it doesn't negatively impact other fields, and I didn't notice any issues.

File: src/app/shared/form/form.service.ts
Comment (atarix83) [inline src/app/shared/form/form.service.ts:165]: I'd restrict this approach to only concat group model (you can use `FormBuilderService.isConcatGroup`). Because a form group can be made of unrelated fields and in this way we are assigned the error to the sub-fields of a form group without distinction

File: src/app/shared/form/form.service.ts
Comment (atarix83) [inline src/app/shared/form/form.service.ts:186]: the same as the add error method

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/ds-dynamic-input.model.ts
Comment (atarix83) [inline src/app/shared/form/builder/ds-dynamic-form-ui/models/ds-dynamic-input.model.ts:30]: probably it's better to have a more generic flag to hide a possible error message and not only the required error message

**Review by atarix83 — CHANGES_REQUESTED** [review]
thanks @ybnd 

While reviewing I found out a problem when the concat group is required and repeatable. 
Schermata da 2022-09-28 15-20-18

As you can see the error message is not shown and in the second entry the second input is not marked

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/ds-dynamic-input.model.ts
Comment (ybnd) [reply src/app/shared/form/builder/ds-dynamic-form-ui/models/ds-dynamic-input.model.ts:30]: Indeed, it was already implemented this way so I just changed the name to `hideErrorMessages` for clarity

File: src/app/shared/form/form.service.ts
Comment (ybnd) [reply src/app/shared/form/form.service.ts]: Done.

File: src/app/shared/form/form.service.ts
Comment (ybnd) [reply src/app/shared/form/form.service.ts]: Done.

**Comment by ybnd** [comment]
@atarix83 Was the problem you mentioned above when saving/depositing the form or just "on blur"?

The original issue was regarding save/deposit -- I agree that the "on blur" behaviour is less than ideal right now. 
Yesterday & this morning I tried to get it closer to the save/depost behaviour (show message below the first in the array) but I can't find a decent solution.

At this point it may be better to leave this to someone that's more experienced with `ng-dynamic-forms` as I'm out of my depth & losing a lot of time on this...

**Comment by tdonohue** [comment]
@ybnd : We talked about this in today's meeting & decided that we might be able to move this PR forward as a "quick fix" even if it's not perfect, as it seems *better* than current behavior.  We could then enhance this fix later on in a future release (and future PR) as needed.  @atarix83 said he'd look at whether this was good enough as-is to solve the main bug...in which case we'll try to fit this into 7.4

In the meantime, I notice your recent changes have caused some small test failures.  Could you look into those & get them fixed?  Looks like a few specs need updates.

**Comment by ybnd** [comment]
Updated & rebased to keep related changes close together

**Comment by jtimal** [comment]
Hello @ybnd 

My name is Julian Timal, I took the boldness to check this PR, when you add more than one contributor it does show the required field message at the moment but when you save it does not show all the messages in all the boxes, is this functionality correct?
image
best regards.

**Comment by ybnd** [comment]
@jtimal thanks for taking the time, but these are not the fields in question -- what you're seeing is default DSpace behaviour
This PR concerns `name` fields that include separate inputs for family/given name (see the screenshots above); these were not working like the regular `onebox` fields before, the intent was to provide a (partial) fix

**Comment by jtimal** [comment]
> @jtimal thanks for taking the time, but these are not the fields in question -- what you're seeing is default DSpace behaviour
> 
> This PR concerns `name` fields that include separate inputs for family/given name (see the screenshots above); these were not working like the regular `onebox` fields before, the intent was to provide a (partial) fix
> 
> To test this PR you'd need to run a REST instance with custom submission forms as outlined in the original issue: #1353

Thanks for your answer, I thought it was for any field anyway I checked and it works fine.
image

**Review by tdonohue — APPROVED** [review]
👍 Thanks @ybnd !  Tested again today and verified that the issues previously reported by @atarix83 are solved (Now, if you have two empty `name` fields, the first one is highlighted in red and the error shown below that one, just like the screenshot from @jtimal above).

As this PR has two approvals (myself and @jtimal ), I'm going to merge this as-is for 7.4.  There may be more improvements we can make in the future, but this PR is at least much better behavior than current `main`.

**Pull request merged**
